numpy/add_newdocs.py

```diff
@@ -2926,10 +2926,14 @@ def luf(lamdaexpr, *args, **kwargs):
     """
     Tuple of array dimensions.
 
-    Notes
-    -----
-    May be used to "reshape" the array, as long as this would not
-    require a change in the total number of elements
+    The shape property is usually used to get the current shape of an array,
+    but may also be used to reshape the array in-place by assigning a tuple of
+    array dimensions to it.  As with `numpy.reshape`, one of the new shape
+    dimensions can be -1, in which case its value is inferred from the length
+    of the array and remaining dimensions. Reshaping an array in-place is not
+    recommended; not only may it fail, but it also affects any code holding a
```

Comment:
I think elaborating on `may it fail` would be good, since I think that is the one reason _to_ use it. Perhaps `Unlike ndarray.reshape, this throws an exception if the reshape would require a copy`.